### DISCUSSION TITLE
Detect Steamdeck and make (mostly minor) changes

### DISCUSSION THREAD

**Discussion by Krutonium** [OpenRCT2/OpenRCT2#20911] [Feature requests]
The Steamdeck is a great way to play OpenRCT2 - With some tweaks.

The primary tweaks that would make OpenRCT2 a great experience out of the box are as follows:

Is it possible to use Steam Input? If yes, we should configure the controls like this, otherwise maybe pop up a dialog to prompt the player to switch to the input layout? I can upload it to Steam and there's probably a way to share it directly via link.

 - Left Trigger - Right Click
 - Right Trigger - Left Click
 - Left Bumper - Rotate World
 - Right Bumper - Rotate World CounterClockwise
 - Both Touchpads as Mouse
 - Left Touchpad right clicks
 - Right Touchpad left clicks
 - A Button - Left Click
 - B Button - Backspace
 - X Button - Right Click
 - Y Button - Rotate Build
 - Left Joystick - Move World
 - Right Joystick - Thus far I've not really found a good use for it
 - Touchpad Speed 50%
 
 - The right hamburger menu I used for pause; but the default binding for it right now isn't available to be bound via Steam, so I bound it to MediaPause
 - The left... idk, 2 boxes on a button button I bound to MediaPlay to Save the Game
 
 
 (This is how I've set it up quite comfortably, but I'm sure there are probably ideas for improvements that could be made, and should also be usable as a guide for anyone setting up their own controls as well I figure)
 
 Ideally we should also set the GUI scaling to 1.75 - It's as big as we can get for the buttons without having the UI go off the sides of the screen in various places, and without overlap on the top bar. Also, switching to the Traffic Light system is a Must imo, the flag system for opening and closing stuff is difficult to use.
 
 Also there needs to be something done when a Text dialog pops up, as right now you have to manually press Steam + B to bring up the keyboard, but in the original RCT and RCT2 versions, the native dialog prompts it to come up automatically.
 
 As an aside since this is on the SteamDeck, if we could also modify the flatpack to have access to `/run/media` by default that would be fantastic, since if RCT1 and 2 are installed on an SD Card they're otherwise inaccessable.

**Comment by Krutonium** [comment]
https://github.com/OpenRCT2/OpenRCT2/assets/3945538/9be9ec5b-6e95-4a5e-8bd9-11e65f83d13e



Here it is with my tweaks. You can see the difficulties, but also the wins. (Ignore any Audio/Video desync, that's not present live, just in the recording)

**Comment by ocalhoun6** [comment]
>Right Joystick - Thus far I've not really found a good use for it

Maybe zoom in/zoom out? Can it be configured to do that?

----

Did you test the different rendering methods?

Since the Steam Deck is fairly standardized, you could set it to use the best-performing rendering method by default.

(I know on my Linux PC, changing the rendering method from default to OpenGL *massively* improved performance, making it a lot smoother.)

**Reply by Krutonium** [reply]
I'd have to grab an intense park to really test that; I didn't find any major performance or power usage differences between the different front ends; The biggest change I made was underclocking the decks display to 40 Hz; And even that only netted like, 30 minutes on top of the already estimated 6 and a half hours from full charge. Basically it runs at 60 FPS pretty well constantly.

**Reply by Krutonium** [reply]
I managed to get both lagging, and OpenGL is slower, but it takes a monster of a park to get to that point.
